src/pykeen/triples/statement_factory.py

```diff
@@ -0,0 +1,600 @@
+    mapped_statements[:, 1::2] = relation_column
+
+    # Note: Unique changes the order of the triples
+    # Note: Using unique means implicit balancing of training samples
+    unique_mapped_statements = np.unique(ar=mapped_statements, axis=0)
+    return torch.tensor(unique_mapped_statements, dtype=torch.long)
+
+
+class StatementFactory(KGInfo):
+    """A mix of CoreTriplesFactory and TriplesFactory, WIP"""
+
+    triples_file_name: ClassVar[str] = "numeric_statements.tsv.gz"
+    base_file_name: ClassVar[str] = "base.pth"
+
+    def __init__(
+        self,
+        mapped_statements: Union[MappedTriples, np.ndarray, torch.tensor],
+        entity_to_id: Dict,
+        relation_to_id: Dict,
+        create_inverse_triples: bool = False,
+        metadata: Optional[Mapping[str, Any]] = None,
+        max_num_qualifier_pairs: int = None,
+    ) -> None:
+        """
+        Create the statement factory.
+
+        :param mapped_statements: shape: (n, max_num_qualifier_pairs)
+            A matrix where each row is a statement (h, r, t, {qr1, qe2}_i).
+        :param num_entities:
+            The number of entities.
+        :param num_relations:
+            The number of relations.
+        :param create_inverse_triples:
+            Whether to create inverse triples.
+        :param metadata:
+            Arbitrary metadata to go with the graph
+        :param max_num_qualifier_pairs:
+            Maximum number of qualifier pairs to load
+
+        :raises TypeError:
+            if the mapped_statements are of non-integer dtype
+        :raises ValueError:
+            if the mapped_statements are of invalid shape
+        """
+        num_entities = len(entity_to_id)
+        num_relations = len(relation_to_id)
+        super().__init__(
+            num_entities=num_entities,
+            num_relations=num_relations,
+            create_inverse_triples=create_inverse_triples,
+        )
+        self.entity_to_id = entity_to_id
+        self.relation_to_id = relation_to_id
+        # ensure torch.Tensor
+        mapped_statements = torch.as_tensor(mapped_statements)
+        # input validation
+        if mapped_statements.ndim != 2 or mapped_statements.shape[1] != (max_num_qualifier_pairs * 2 + 3):
+            raise ValueError(
+                f"Invalid shape for mapped_triples: {mapped_statements.shape}; "
+                f"must be (n, 3 + max_num_qualifier_pairs * 2)"
+            )
+        if mapped_statements.is_complex() or mapped_statements.is_floating_point():
+            raise TypeError(f"Invalid type: {mapped_statements.dtype}. Must be integer dtype.")
+        # always store as torch.long, i.e., torch's default integer dtype
+        self.mapped_statements = mapped_statements.to(dtype=torch.long)
+        if metadata is None:
+            metadata = dict()
+        self.metadata = metadata
+        self.max_num_qualifier_pairs = max_num_qualifier_pairs
+        self.paddix_idx = entity_to_id["__padding__"]
```

Comment:
```suggestion
        self.padding_idx = entity_to_id["__padding__"]
```

Should we mention somewhere that entity_to_id needs to have this entity?

Reply:
another comment: there is a constant for this in another file (named `STATEMENT_PADDING`)